lib/fastlane/plugin/ddg_apple_automation/actions/tds_perf_test.rb

```diff
@@ -0,0 +1,145 @@
+require "fastlane/action"
+require "fastlane_core/configuration/config_item"
+require "octokit"
+require_relative "../helper/asana_helper"
+require_relative "../helper/ddg_apple_automation_helper"
+require_relative "../helper/git_helper"
+
+module Fastlane
+  module Actions
+    class TdsPerfTestAction < Action
+      # Define platform-specific constants
+      IOS_TEST_PARAMS = {
+        ut_file_name: "ios-tds.json",
+        ut_url: "https://staticcdn.duckduckgo.com/trackerblocking/v5/current/ios-tds.json",
+        ref_file_name: "trackerData.json",
+        ref_url: "https://raw.githubusercontent.com/duckduckgo/apple-browsers/refs/heads/main/iOS/Core/"
+      }.freeze
+
+      MAC_TEST_PARAMS = {
+        ut_file_name: "macos-tds.json",
+        ut_url: "https://staticcdn.duckduckgo.com/trackerblocking/v6/current/",
+        ref_file_name: "trackerData.json",
+        ref_url: "https://raw.githubusercontent.com/duckduckgo/apple-browsers/refs/heads/main/macOS/DuckDuckGo/ContentBlocker/"
+      }.freeze
+
+      def self.run(params)
+        UI.message("Starting TDS Performance Testing...")
+
+        # Determine platform and set default parameters if needed
+        platform = lane_context[SharedValues::PLATFORM_NAME]
+        default_params = platform == :ios ? IOS_TEST_PARAMS : MAC_TEST_PARAMS
+
+        # Use provided parameters or defaults
+        ut_file_name = params[:ut_file_name] || default_params[:ut_file_name]
+        ut_url = params[:ut_url] || default_params[:ut_url]
+        ref_file_name = params[:ref_file_name] || default_params[:ref_file_name]
+        ref_url = params[:ref_url] || default_params[:ref_url]
+
+        UI.message("Using TDS parameters for #{platform} platform:")
+        UI.message("  Under-test file: #{ut_file_name}")
+        UI.message("  Under-test URL: #{ut_url}")
+        UI.message("  Reference file: #{ref_file_name}")
+        UI.message("  Reference URL: #{ref_url}")
+
+        # Create temporary directory
+        tmp_dir = "#{ENV.fetch('TMPDIR', nil)}/tds-perf-testing"
```

Comment:
Nit: we could use Ruby API for creating and clearing temporary directory. That said, I can’t recall the correct API off the top of my head :) It’s fine either way as long as it works 👍